modules/api/src/main/java/org/apache/ignite/sql/Session.java

```diff
@@ -0,0 +1,143 @@
+import org.apache.ignite.tx.Transaction;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * SQL Session provides methods for query execution.
+ *
+ * <p>Session is a stateful object and holds setting that intended to be used as defaults for the new queries. Session is thread-safe and
+ * can be reused and shared between threads, which is normal for asynchronous and reactive flow, however, modifying session state from
+ * different threads may lead to the race condition.
+ */
+public interface Session extends AsyncSession, ReactiveSession {
+    /** Default schema name. */
+    String DEFAULT_SCHEMA = "PUBLIC";
+
+    /**
+     * Sets default query timeout.
+     *
+     * @param timeout Query timeout value.
+     * @param timeUnit Timeunit.
+     */
+    void defaultTimeout(long timeout, @NotNull TimeUnit timeUnit);
+
+    /**
+     * Return default query timeout.
+     *
+     * @param timeUnit Timeunit to convert timeout to.
+     * @return Default query timeout in the given timeunit.
+     */
+    long defaultTimeout(@NotNull TimeUnit timeUnit);
+
+    /**
+     * Sets default schema for the session, which the queries will be executed with. The default schema is used for query planning to
+     * resolve table names to their canonical names.
+     *
+     * @param schema Default schema.
+     */
+    void defaultSchema(@NotNull String schema);
+
+    /**
+     * Returns session default schema.
+     *
+     * <p>Default value is {@link #DEFAULT_SCHEMA}.
+     *
+     * @return Session default schema.
+     * @see #defaultSchema(String)
+     */
+    @NotNull String defaultSchema();
+
+    /**
+     * Executes single SQL query.
+     *
+     * @param transaction Transaction to execute the query within or {@code null}.
+     * @param query SQL query template.
+     * @param arguments Arguments for the template (optional).
+     * @return SQL query results set.
+     * @throws SqlException If failed.
+     */
+    @NotNull ResultSet execute(@Nullable Transaction transaction, @NotNull String query, @Nullable Object... arguments);
+
+    /**
+     * Executes single SQL statement.
+     *
+     * @param transaction Transaction to execute the statement within or {@code null}.
+     * @param statement SQL statement to execute.
+     * @param arguments Arguments for the statement.
+     * @return SQL query results set.
+     */
+    @NotNull ResultSet execute(@Nullable Transaction transaction, @NotNull Statement statement, @Nullable Object... arguments);
+
+    /**
+     * Executes batched SQL query.
+     *
+     * @param transaction Transaction to execute the statement within or {@code null}.
+     * @param query SQL query template.
+     * @param batch List of batch rows, where each row is a list of statement arguments.
+     * @return SQL query results set.
+     */
+    @NotNull int[] executeBatch(@Nullable Transaction transaction, @NotNull String query, @NotNull List<List<@Nullable Object>> batch);
```

Comment:
will it have the same semantics as prepared statement which runs multiple times for list of arguments?